### PR TITLE
[Hub Generated] Review request for Microsoft.Insights to add version stable/2022-06-01

### PR DESCRIPTION
**Pull request by herreraj-ms**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/herreraj-ms/azure-rest-api-specs/herreraj%2Famcs%2F2022-06-01...main/).
## ARM API Information (Control Plane)

Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.

The new api-version is already present in all regions behind an AFEC flag

  4. When do you expect to publish the swagger? Please provide date or, the date is not yet available, the month.

As soon as possible. We are moving the new api-version to GA, but there are clients that already using it.

  6. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by herreraj-ms** [comment]
@ArcturusZhang 

The previous version is this one: https://github.com/Azure/azure-rest-api-specs/tree/main/specification/monitor/resource-manager/Microsoft.Insights/preview/2021-09-01-preview

But there was another RP that created the version 2022-06-01 and I think that why the breaking changes validation is not passing.

Can you advise me how to fix this issue? or can it be bypass?

File: specification/monitor/resource-manager/readme.csharp.md [hunk 1]
```diff
@@ -22,6 +22,11 @@ csharp:
   clear-output-folder: true
 ```
 
+``` yaml yaml $(csharp) && $(track2)
+modelerfour:
+  lenient-model-deduplication: true
+```
```
Comment (ArcturusZhang) [inline specification/monitor/resource-manager/readme.csharp.md:28]: Please do not add this flag. DotNet track 2 monitor SDK has already GA, we never use this flag in GA libraries (we should not).
Whatever failures we have in generating monitor SDK, we should resolve its root cause. For instance, rename the one of the duplicated schemas, or we at least could resolve those duplicated schema in configuration in dotnet SDK repo.

File: specification/monitor/resource-manager/readme.go.md [hunk 2]
```diff
@@ -48,6 +48,7 @@ batch:
   - tag: package-2021-05-metrics
   - tag: package-2021-08-scheduledqueryrules
   - tag: package-2022-02-01-preview-only
+  - tag: package-2023-02-01-only
```
Comment (ArcturusZhang) [inline specification/monitor/resource-manager/readme.go.md:51]: We no longer support Go track 1 SDK, therefore no changes should be made for this file. Could we please revert this change?

**Comment by herreraj-ms** [comment]
@ArcturusZhang I think this is a false positive: The situation is adding new specs to a folder that it's already created, this should be a normal situation since azure monitor consist of multiple resource providers.

Our API has been in use for several months with several clients. Now we need to move it to GA so they can have docs/SDKs/Tooling.

What do you need from my side to approve the PR?

**Review by ArcturusZhang — CHANGES_REQUESTED** [review]
Please update the new tag you added according to recent changes: https://github.com/Azure/azure-rest-api-specs/pull/22939
This is the bad part that everyone adds a new tag whenever he/she is adding swagger files. You will quickly lose track for those versions because we have a lot.

**Review by ArcturusZhang — APPROVED** [review]

File: specification/monitor/resource-manager/readme.md [hunk 3]
```diff
@@ -31,12 +31,56 @@ title: MonitorClient
...
+- Microsoft.Insights/stable/2016-03-01/alertRules_API.json
+- Microsoft.Insights/stable/2016-03-01/logProfiles_API.json
+- Microsoft.Insights/preview/2021-05-01-preview/diagnosticsSettings_API.json
+- Microsoft.Insights/preview/2021-05-01-preview/diagnosticsSettingsCategories_API.json
+- Microsoft.Insights/stable/2023-01-01/actionGroups_API.json
+- Microsoft.Insights/stable/2015-04-01/activityLogs_API.json
+- Microsoft.Insights/stable/2015-04-01/eventCategories_API.json
+- Microsoft.Insights/stable/2015-04-01/tenantActivityLogs_API.json
+- Microsoft.Insights/stable/2018-01-01/metricDefinitions_API.json
+- Microsoft.Insights/stable/2018-01-01/metrics_API.json
```
Comment (tadelesh) [inline specification/monitor/resource-manager/readme.md:69]:
```suggestion
- Microsoft.Insights/stable/2021-05-01/metricDefinitions_API.json
- Microsoft.Insights/stable/2021-05-01/metrics_API.json
```

Could you help to change metrics to the latest version?

File: specification/monitor/resource-manager/readme.md [hunk 4]
```diff
@@ -31,12 +31,56 @@ title: MonitorClient
...
+### Tag: package-2023-02-01-only
+
+These settings apply only when `--tag=package-2023-02-01-only` is specified on the command line
+
+``` yaml $(tag) == 'package-2023-02-01-only'
+input-file:
+- Microsoft.Insights/stable/2022-06-01/dataCollectionEndpoints_API.json
+- Microsoft.Insights/stable/2022-06-01/dataCollectionRuleAssociations_API.json
+- Microsoft.Insights/stable/2022-06-01/dataCollectionRules_API.json
+```
```
Comment (ArcturusZhang) [inline specification/monitor/resource-manager/readme.md:49]: The api-version of your swagger is not `2023-02-01` therefore you cannot name your tag in this way, it makes no sense when a package-yyyy-mm-dd-only tag does not contain any swagger with that specific date.
Since `package-2022-06-01-only` is taken, you could use `package-datacollection-2022-06-01-only` instead to emphasize the api-version.

**Review by ArcturusZhang — CHANGES_REQUESTED** [review]

**Review by tadelesh — APPROVED** [review]

**Review by ArcturusZhang — APPROVED** [review]
